src/main/java/org/wildfly/security/audit/SizeRotatingFileAuditEndpoint.java

```diff
@@ -0,0 +1,202 @@
+            rotate(file);
+        }
+    }
+
+    @Override
+    protected void write(byte[] bytes) throws IOException {
+        super.write(bytes);
+        currentSize += bytes.length;
+    }
+
+    @Override
+    protected void preWrite(Date date) {
+        if (currentSize > rotateSize && maxBackupIndex > 0) {
+            try {
+                final File file = getFile();
+                if (file == null) {
+                    // no file is set; a direct output stream or writer was specified
+                    return;
+                }
+                rotate(file);
+                currentSize = 0;
+            } catch (IOException e) {
+                audit.unableToRotateLogFile(e);
+            }
+        }
+    }
+
+    /**
+     * Moves file to file.1, file.1 to file.2 etc. Removes file.{maxBackupIndex}
+     */
+    private void rotate(final File file) throws IOException {
+        closeStreams();
+        final String suffix = dateTimeFormatter != null ? dateTimeFormatter.format(ZonedDateTime.now()) : "";
+        final Path fileWithSuffix = Paths.get(file.getAbsolutePath() + suffix);
+        Files.deleteIfExists(Paths.get(fileWithSuffix + "." + maxBackupIndex));
+        for (int i = maxBackupIndex - 1; i >= 1; i--) {
+            final Path src = Paths.get(fileWithSuffix + "." + i);
+            if (Files.exists(src)) {
+                final Path target = Paths.get(fileWithSuffix + "." + (i + 1));
+                Files.move(src, target, StandardCopyOption.REPLACE_EXISTING);
+            }
+        }
+        Files.move(file.toPath(), Paths.get(fileWithSuffix + ".1"), StandardCopyOption.REPLACE_EXISTING);
+        setFile(file);
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder extends FileAuditEndpoint.Builder {
+
+        private long rotateSize = 0xa0000L; // 10 MB by default
+        private int maxBackupIndex = 1;
+        private boolean rotateOnBoot;
+        private String suffix;
+
+        ZoneId timeZone = ZoneId.systemDefault();
+
+        Builder() {
+            super();
+        }
+
+        /**
+         * Set the log file size the file should rotate at.
+         *
+         * @param rotateSize the size the file should rotate at
+         * @return this builder.
+         */
+        public Builder setRotateSize(long rotateSize) {
+            this.rotateSize = rotateSize;
+
+            return this;
+        }
+
+        /**
+         * Sets the suffix to be appended to the file name during the file rotation. The suffix does not play a role in
+         * determining when the file should be rotated.
+         * <p/>
+         * The suffix must be a string understood by the  {@link java.time.format.DateTimeFormatter}.
+         * <p/>
+         * <b>Note:</b> Any files rotated with the suffix appended will not be deleted. The {@link #setMaxBackupIndex(int)
+         * maxBackupIndex} is not used for files with a suffix.
```

Comment:
We should fix this documentation as it's not quite correct. Files with the same date suffix will be deleted.